src/Backends/Housekeeping.vala

```diff
@@ -133,16 +119,59 @@ public class SettingsDaemon.Backends.Housekeeping : GLib.Object {
             return;
         }
 
-        // See https://www.freedesktop.org/software/systemd/man/tmpfiles.d.html for details
-        // Results in a config line like:
-        // e "/home/david/Downloads" - - - 30d
-        string config = "e \"%s\" - - - %dd".printf (downloads_folder, downloads_cleanup_days);
-
         FileOutputStream os = config_stream.output_stream as FileOutputStream;
         try {
-            yield os.write_all_async (config.data, GLib.Priority.DEFAULT, null, null);
+            yield os.write_all_async (cleanup_config.data (), Priority.DEFAULT, null, null);
         } catch (Error e) {
             warning ("Unable to write systemd-tmpfiles config: %s", e.message);
         }
     }
+
+    private class CleanupConfig : Object {
+        public bool clean_downloads { private get; public construct; }
+        public bool clean_screenshots { private get; public construct; }
+        public int clean_after_days { private get; public construct; }
+
+        public bool is_disabled { get {
+            return (!clean_downloads && !clean_screenshots) || clean_after_days < 1;
+        }}
+
+        public CleanupConfig (Settings settings) {
+            Object (
+                clean_downloads: settings.get_boolean ("cleanup-downloads-folder")
+                    && downloads_are_not_home (),
+                clean_screenshots: settings.get_boolean ("cleanup-screenshots-folder"),
+                clean_after_days: settings.get_int ("old-files-age")
+            );
+        }
+
+        private static bool downloads_are_not_home () {
+            var home_dir = Environment.get_home_dir ();
+            var home = File.new_for_path (home_dir);
+            var downloads_dir = Environment.get_user_special_dir (UserDirectory.DOWNLOAD);
+            var downloads = File.new_for_path (downloads_dir);
+            return !home.equal (downloads);
+        }
+
+        public uint8[] data () {
+            // See https://www.freedesktop.org/software/systemd/man/tmpfiles.d.html for details
+            // Results in a config line like:
+            // e "/home/david/Downloads" - - - 30d
+            var template = "e \"%s\" - - - %dd";
+            string[] lines = {};
+
+            if (clean_downloads) {
+                var downloads_dir = Environment.get_user_special_dir (UserDirectory.DOWNLOAD);
+                lines += template.printf (downloads_dir, clean_after_days);
+            }
+
+            if (clean_screenshots) {
+                var pictures_dir = Environment.get_user_special_dir (UserDirectory.PICTURES);
+                var screenshots_dir = Path.build_filename (pictures_dir, _("Screenshots"));
```

Comment:
This `_("Screenshots")` here bothers me. [The Screenshot app itself uses a translated directory name](https://github.com/elementary/screenshot/blob/master/src/Application.vala#L21). That's why I used a translated string here as well. However, that leaves room for missing, or just inconsistent, translations (in either app) which will lead to incorrect cleanup behavior.

I guess this could be avoided if we somehow synced translations, but I'm not sure if that's possible (or how to do it). @danirabbit do you have any suggestions?

Reply:
I think there's a way to include the gettext domain here and use the translations from screenshot. I think we should actually be using the GNOME Settings daemon gettext domain here maybe? @tintou would probably know what to do about that

Reply:
I did some digging:

1. I did not find a translation for `"Screenshots"` in [GNOME settings daemon](https://gitlab.gnome.org/GNOME/gnome-settings-daemon/-/blob/master/po/en_GB.po). And if I'm reading its code correctly, I thing [GNOME's default screenshot utility](https://gitlab.gnome.org/GNOME/gnome-screenshot/-/blob/master/src/screenshot-filename-builder.c#L101) saves images straight into `~/Pictures`, without dedicated screenshots sub-directory.
2. I noticed that we have screenshot saving implemented in 2 places: [in Gala](https://github.com/elementary/gala/blob/master/src/ScreenshotManager.vala#L273) and in [standalone Screenshot app](https://github.com/elementary/screenshot/blob/master/src/Application.vala#L21).

The implementation between Gala and Screenshot app differs slightly: the app allows users to override the directory and will remember the new destination for later. On the other hand Gala implementation always uses the translated `~/Pictures/Screenshots` directory.

Since Gala implementation looked more "stable" to me I opted for reusing that translation for this housekeeping feature. This effectively means that we'll "leak" screenshot images taken with the Screenshot app in 2 cases:
1. If user overrides the default directory.
2. If Screenshot app's translation differs from Gala's.

The 2nd one could be resolved by updating the Screenshot app to reuse Gala translations similar to how it's done here. The 1st issue, on the other hand, would be tricky to support. For example user might want to save screenshots into their home directory. I would be wrong to clean up all images from their home dir because of this.

What do you think? Is this implementation good enough?